shoop/campaigns/migrations/0006_effects.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import shoop.core.fields
+
+def forwards_func(apps, schema_editor):
+    CatalogCampaign = apps.get_model("campaigns", "CatalogCampaign")
+    CatalogCampaignEffect = apps.get_model("campaigns", "CatalogCampaignEffect")
+    CatalogDiscountAmount = apps.get_model("campaigns", "CatalogDiscountAmount")
+    CatalogDiscountPercentage = apps.get_model("campaigns", "CatalogDiscountPercentage")
+
+    BasketCampaign = apps.get_model("campaigns", "BasketCampaign")
+    BasketCampaignEffect = apps.get_model("campaigns", "BasketCampaignEffect")
+    BasketDiscountAmount = apps.get_model("campaigns", "BasketDiscountAmount")
+    BasketDiscountPercentage = apps.get_model("campaigns", "BasketDiscountPercentage")
+
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+
+    db_alias = schema_editor.connection.alias
+
+    for c in BasketCampaign.objects.using(db_alias).all():
+        if not c.discount_amount_value and not c.discount_percentage:
+            return
+
+        if c.discount_amount_value:
+            new_ct = ContentType.objects.get_for_model(BasketDiscountAmount)
+            e = BasketDiscountAmount()
+            e.discount_amount = c.discount_amount_value
+        else:
+            new_ct = ContentType.objects.get_for_model(BasketDiscountPercentage)
+            e = BasketDiscountPercentage()
+            e.discount_percentage = c.discount_percentage
+        e.campaign = c
+        e.save()
+
+        BasketCampaignEffect.objects.filter(polymorphic_ctype__isnull=True, campaign_id=c.id).update(polymorphic_ctype=new_ct)
+
+    for c in CatalogCampaign.objects.using(db_alias).all():
```

Comment:
Looks like you are using exactly same logic to handle both BasketCampaigns and CatalogCampaigns. I think you should make this forwards func little bit prettier.